app/models/daily_report.rb

```diff
@@ -16,14 +16,26 @@ class DailyReport < ApplicationRecord
               maximum: Settings.daily_report.max_content_length
             }
   validate :one_report_per_day, on: :create
-  validate :check_user_course_association, on: :create
+  validate :check_user_course_association, on: :create,
+            if: -> {course_id.present?}
 
   # Scopes
   scope :completed, -> {where(is_done: true)}
   scope :pending, -> {where(is_done: false)}
   scope :recent, -> {order(created_at: :desc)}
   scope :by_user, ->(user) {where(user:)}
   scope :by_course, ->(course) {where(course:)}
+  scope :by_courses, ->(course_ids) {where(course_ids:)}
+  scope :on_day, (lambda do |date|
+                    if date.present?
+                      processed_date = Date.strptime(date, Settings.params.date)
+                      where(created_at: processed_date
+                      .beginning_of_day..processed_date.end_of_day)
+                    end
+                  end)
```

Comment:
```suggestion
  scope :on_day, (lambda do |date|
    return if date.blank?
    
    processed_date = Date.strptime(date, Settings.params.date)
    where(created_at: processed_date.beginning_of_day..processed_date.end_of_day)
  end)
```
Lùi code lại cho gọn e nha, viết được dài hơn